### DISCUSSION TITLE
Lightweight loading of meshes in Android

### DISCUSSION THREAD

**Discussion by AbelToy** [google/filament#6981] [Q&A] [answered]
I would like to load some meshes in Android, without having to depend on the gltfio library, since it adds about 2MB to the app size, and I don't need most of gltf features.

There is a much more lighterweight library in the source code called filameshio, with a MeshReader class that can read filamesh files directly.

However, there are currently no bindings for that library in filamesh Android.

Additionally, the Android samples provide a custom implementation (called MeshLoader). Unfortunately, that class only exists in the samples and not in the library itself, and thus I would need to copy it to my project's sources in order to use it, and I would like to avoid that in order to reduce the maintenance needed.

I was wondering whether there are any plans for adding bindings / support to load filamesh files directly on Android, or if just copying MeshLoader is the expected route here. Additionally, I was wondering if this use case is actually what I should do, or if it's just expected for users to depend on the gltfio library.

Best regards and thanks for the great work in this fantastic library! 😄

**Comment by romainguy** [comment, marked as answer]
Copying MeshReader and customizing it if necessary is the expected use case. We purposefully don't want to provide/support mesh formats as Filament is intended to be purely a renderer. "Filamesh" is not a format that evolves so copying its reader will it increase maintenance burden on your end. Filamesh files are designed to be both small and extremely efficient to read (there's pretty much 0 parsing, the meshes can be loaded as encoded in the file which means you can mmap the file).

**Reply by AbelToy** [reply]
Thanks!! That sounds perfect for my use case then.
